src/vanilla.ts

```diff
@@ -9,16 +9,19 @@ export type StateListener<T> = (state: T, previousState: T) => void
  * @deprecated Use `StateListener<T>` instead of `StateSliceListener<T>`.
  */
 export type StateSliceListener<T> = (slice: T, previousSlice: T) => void
-export interface Subscribe<T extends State> {
-  (listener: StateListener<T>): () => void
-}
+export type Subscribe<T extends State> = {
+  _(listener: StateListener<T>): () => void
+}['_']
```

Comment:
This can be left as it is as mentioned [here](https://github.com/pmndrs/zustand/pull/1029#issuecomment-1165717355).

```suggestion
export interface Subscribe<T extends State> {
  (listener: StateListener<T>): () => void
}
```